### DISCUSSION TITLE
Error saving request: PrismaClientKnownRequestError: Invalid `prisma.request.create()` invocation: The table `public.Request` does not exist in the current database.

### DISCUSSION THREAD

**Discussion by kumarshivesh** [prisma/prisma#24607] [Q&A]
### Question

I made a REST client application using Next.js, enabling users to perform various HTTP requests (GET, POST, PUT, DELETE) with real-time response rendering and historical request logging using Prisma and PostgreSQL. Eveything is working fine in the locally in my laptop. But when I tried to make any of HTTP requests on the deployed link then I am getting this following error in the Render log (I deployed this app on Render):
```
Error saving request: PrismaClientKnownRequestError: 
Invalid `prisma.request.create()` invocation:
The table `public.Request` does not exist in the current database.
    at In.handleRequestError (/opt/render/project/src/node_modules/@prisma/client/runtime/library.js:122:6877)
    at In.handleAndLogRequestError (/opt/render/project/src/node_modules/@prisma/client/runtime/library.js:122:6211)
    at In.request (/opt/render/project/src/node_modules/@prisma/client/runtime/library.js:122:5919)
    at async l (/opt/render/project/src/node_modules/@prisma/client/runtime/library.js:127:11167)
    at async c (/opt/render/project/src/.next/server/app/api/store-request/route.js:1:652)
    at async /opt/render/project/src/node_modules/next/dist/compiled/next-server/app-route.runtime.prod.js:6:36258
    at async eR.execute (/opt/render/project/src/node_modules/next/dist/compiled/next-server/app-route.runtime.prod.js:6:26874)
    at async eR.handle (/opt/render/project/src/node_modules/next/dist/compiled/next-server/app-route.runtime.prod.js:6:37512)
    at async doRender (/opt/render/project/src/node_modules/next/dist/server/base-server.js:1377:42)
    at async cacheEntry.responseCache.get.routeKind (/opt/render/project/src/node_modules/next/dist/server/base-server.js:1599:28) {
  code: 'P2021',
  clientVersion: '5.15.1',
  meta: { modelName: 'Request', table: 'public.Request' }
}
```
### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

generator client {
  provider = "prisma-client-js"
}

model Request {
  id          Int      @id @default(autoincrement())
  method      String
  url         String
  requestBody String?
  responseBody String?
  createdAt   DateTime @default(now())
}
```

```ts
// Add any relevant Prisma Client queries here

import { PrismaClient } from '@prisma/client';

const prisma = new PrismaClient();

// Create a Request
async function createRequest() {
  const request = await prisma.request.create({
    data: {
      method: 'GET',
      url: 'http://example.com',
      requestBody: '{}',
      responseBody: '{}'
    }
  });
  console.log(request);
}

createRequest()
  .catch(e => {
    throw e;
  })
  .finally(async () => {
    await prisma.$disconnect();
  });

// Fetch All Requests
async function getRequests() {
  const requests = await prisma.request.findMany();
  console.log(requests);
}

getRequests()
  .catch(e => {
    throw e;
  })
  .finally(async () => {
    await prisma.$disconnect();
  });

// Delete a Request by ID
async function deleteRequest(id: number) {
  const request = await prisma.request.delete({
    where: { id }
  });
  console.log(request);
}

deleteRequest(1)
  .catch(e => {
    throw e;
  })
  .finally(async () => {
    await prisma.$disconnect();
  });

```

- OS:  Windows
- Database:  PostgreSQL
- Node.js version:  20.13.1

- Run `prisma -v` to see your Prisma version and paste it
- prisma                  : 5.15.1

**Comment by ludralph** [comment]
Hi @kumarshivesh 👋 

Thank you for raising this question.

Can you confirm that you have run migrations in production by using the command `npx prisma migrate deploy`?

**Reply by kumarshivesh** [reply]
Hi @ludralph,

Yes, I ran migrations in production by using the command `npx prisma migrate deploy`.

I tried the following steps to resolve this problem but unfortunately it did not work.

```
# Step 1: Uninstall Prisma Client
npm uninstall @prisma/client

# Step 2: Delete the `.prisma` directory manually from `node_modules`

# Step 3: Reinstall Prisma Client
npm install @prisma/client

# Step 4: Generate Prisma Client
npx prisma generate

# Step 5: Apply Migrations Locally
npx prisma migrate dev --name init

# Step 6: Push Changes to GitHub
git add .
git commit -m "Ensure Prisma migrations are up to date"
git push origin main

# Step 7: Apply Migrations on Render
npx prisma migrate deploy
```

**Reply by ludralph** [reply]
Hi @kumarshivesh 

It's likely that you haven't run your Prisma migrations on the production database. The error The table `public.Request` does not exist in the current database suggests that the schema hasn't been applied to your production database.

To fix this, you need to run Prisma migrations on your production database. Here are the steps you can follow:

a. First, make sure your `DATABASE_URL` environment variable on Render is correctly set to your production database URL.
b. In your Render dashboard, you can add a build command to run the migrations. Add this to your build commands:
```
npx prisma generate && npx prisma migrate deploy
```
The prisma generate command will generate the Prisma client, and prisma migrate deploy will apply all pending migrations to your production database.

If you're using Prisma Migrate, make sure you have migration files in your `prisma/migrations` folder. If not, you may need to create an initial migration locally using npx prisma migrate dev and commit these files to your repository.

After setting up the build command, redeploy your application on Render.

If you've already tried these steps and are still encountering issues, you might want to check:

- If your DATABASE_URL environment variable on Render is correctly set to your production database.
- If you have the necessary permissions to create tables in your production database.
- If there are any network restrictions preventing your Render deployment from accessing your database.

**Reply by ludralph** [reply]
You can also take a look at this guide to [deploy to Render](https://www.prisma.io/docs/orm/prisma-client/deployment/traditional/deploy-to-render)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!;
